client/customloaders/customfileloader/index.js

```diff
@@ -0,0 +1,59 @@
+var Promise = require('bluebird');
+var fse = Promise.promisifyAll(require('fs-extra'));
+var loaderUtils = require('loader-utils');
+var mime = require('mime');
+var path = require('path');
+
+module.exports = function (content) {
+  this.cacheable && this.cacheable();
```

Comment:
This reminds me of one of the most  famous JavaScript discussion in history:

https://github.com/twbs/bootstrap/issues/3057

Arguable, but I'd prefer using `if`:

```javascript
if (this.cacheable) {
  this.cacheable();
}
```

or even:

```javascript
if (typeof this.cacheable === 'function') {
  this.cacheable();
}